cylc/flow/task_pool.py

```diff
@@ -1114,7 +1114,7 @@ def spawn_tasks(self, items, failed, non_failed):
             submit_num = submit_nums.get(key, 0)
 
             # This the upstream target task:
-            itask = TaskProxy(taskdef, point, self.config.initial_point)
+            itask = TaskProxy(taskdef, self.config.initial_point, point)
```

Comment:
Random comment from lurker: this looks like maybe a good place to have those `__init__(self, *, initial_point, point)`, to force users of the API to specify the parameter name, avoiding accidentally switching parameters?

Reply:
Ha ha good spotting. Yeah, but we should probably do that as as part of a coherent effort to make the entire API more robust (particularly once we have an API intended for external use).  In this case the functional tests detected my error, at least.